src/CardUtils.res

```diff
@@ -581,3 +581,85 @@ let postalRegex = (postalCodes: array<PostalCodeType.postalCodes>, ~country=?, (
   let countryPostal = Utils.getCountryPostal(country, postalCodes)
   countryPostal.regex == "" ? "" : countryPostal.regex
 }
+
+let getCardDetailsFromCardProps = cardProps => {
+  let defaultCardProps = (
+    None,
+    _ => (),
+    "",
+    _ => (),
+    _ => (),
+    React.useRef(Js.Nullable.null),
+    <> </>,
+    "",
+    _ => (),
+    0,
+  )
+
+  switch cardProps {
+  | Some(cardProps) => cardProps
+  | None => defaultCardProps
+  }
+}
+
+let getExpiryDetailsFromExpiryProps = expiryProps => {
+  let defaultExpiryProps = (
+    None,
+    _ => (),
+    "",
+    _ => (),
+    _ => (),
+    React.useRef(Js.Nullable.null),
+    _ => (),
+    "",
+    _ => (),
+  )
+
+  switch expiryProps {
+  | Some(expiryProps) => expiryProps
+  | None => defaultExpiryProps
+  }
+}
+
+let getCvcDetailsFromCvcProps = cvcProps => {
+  let defaultCvcProps = (
+    None,
+    _ => (),
+    "",
+    _ => (),
+    _ => (),
+    _ => (),
+    React.useRef(Js.Nullable.null),
+    _ => (),
+    "",
+    _ => (),
+  )
+
+  switch cvcProps {
+  | Some(cvcProps) => cvcProps
+  | None => defaultCvcProps
+  }
+}
+
+let setRightIconForCvc = (~cardEmpty, ~cardInvalid, ~color, ~cardComplete) => {
+  if cardEmpty {
+    <Icon size=28 name="cvc-empty" />
+  } else if cardInvalid {
+    <div style={ReactDOMStyle.make(~color, ())}>
+      <Icon size=28 name="cvc-invalid" />
+    </div>
+  } else if cardComplete {
+    <Icon size=28 name="cvc-complete" />
+  } else {
+    <Icon size=28 name="cvc-empty" />
+  }
+}
+
+let getCardDetailsHook = (~cvcNumber, ~isCvcValidValue, ~isCVCValid) => {
```

Comment:
Custom hooks needs to start with 'use' by best practice for clear demarcation so nobody uses it like a function.